gluten-uniffle/src/main/java/org/apache/spark/shuffle/gluten/uniffle/UniffleShuffleManager.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.spark.shuffle.ShuffleWriter;
+import org.apache.spark.shuffle.sort.ColumnarShuffleManager;
+
+import org.apache.spark.ShuffleDependency;
+import org.apache.spark.SparkConf;
+import org.apache.spark.SparkEnv;
+import org.apache.spark.TaskContext;
+import org.apache.spark.executor.ShuffleWriteMetrics;
+import org.apache.spark.shuffle.writer.BufferManagerOptions;
+import org.apache.spark.shuffle.writer.UniffleColumnarShuffleWriter;
+import org.apache.spark.shuffle.writer.WriteBufferManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UniffleShuffleManager extends RssShuffleManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(UniffleShuffleManager.class);
+  private final ConcurrentHashMap.KeySetView<Integer, Boolean> columnarShuffleIds =
+      ConcurrentHashMap.newKeySet();
+  private volatile ColumnarShuffleManager _columnarShuffleManager;
+  private static final String glutenShuffleManagerName =
+      "org.apache.spark.shuffle.sort.ColumnarShuffleManager";
+
+  public UniffleShuffleManager(SparkConf conf, boolean isDriver) {
+    super(conf, isDriver);
+  }
+
+  private ColumnarShuffleManager columnarShuffleManager() {
+    if (_columnarShuffleManager == null) {
+      synchronized (this) {
+        if (_columnarShuffleManager == null) {
+          _columnarShuffleManager =
+              initColumnManager(
+                  glutenShuffleManagerName, sparkConf, isDriver());
+        }
+      }
+    }
+    return _columnarShuffleManager;
+  }
+
+  private ColumnarShuffleManager initColumnManager(
+      String name, SparkConf conf, boolean isDriver) {
+    Constructor constructor;
+    ColumnarShuffleManager instance;
+    try {
+      Class klass = Class.forName(name);
+      try {
+        constructor = klass.getConstructor(conf.getClass(), Boolean.TYPE);
+        instance = (ColumnarShuffleManager)constructor.newInstance(conf, isDriver);
+      } catch (NoSuchMethodException var7) {
+        constructor = klass.getConstructor(conf.getClass());
+        instance = (ColumnarShuffleManager)constructor.newInstance(conf);
+      }
+    } catch (Exception e) {
+      throw new RuntimeException("initColumnManager fail");
+    }
+    return instance;
+  }
+
+  private boolean isDriver() {
+    return "driver".equals(SparkEnv.get().executorId());
+  }
+
+  // This method is called in Spark driver side,
+  // and Spark driver will make some decision according to coordinator,
+  // e.g. determining what RSS servers to use.
+  // Then Spark driver will return a ShuffleHandle and
+  // pass that ShuffleHandle to executors (getWriter/getReader)
+  @Override
+  public <K, V, C> ShuffleHandle registerShuffle(
+      int shuffleId, ShuffleDependency<K, V, C> dependency) {
+    if (!(dependency instanceof ColumnarShuffleDependency)) {
+      columnarShuffleIds.add(shuffleId);
+      return columnarShuffleManager().registerShuffle(shuffleId, dependency);
```

Comment:
why not super.registerShuffle(shuffleId, dependency);

Reply:
Maybe it's present in next line